bpf/tests/pktgen.h

```diff
@@ -207,24 +211,24 @@ struct iphdr *pktgen__push_iphdr(struct pktgen *builder)
 
 	builder->layers[layer_idx] = PKT_LAYER_IPV4;
 	builder->layer_offsets[layer_idx] = builder->cur_off;
-	builder->cur_off += sizeof(struct iphdr);
+	builder->cur_off += length;
 
 	return layer;
 }
 
-/* Push a IPv4 header with sane defaults onto the packet */
+/* Push a IPv4 header with sane defaults and options onto the packet */
 static __always_inline
 __attribute__((warn_unused_result))
-struct iphdr *pktgen__push_default_iphdr(struct pktgen *builder)
```

Comment:
Could we have a `pktgen__push_default_iphdr_with_opt` and let `pktgen__push_default_iphdr` call it with a 0 second argument, such that we wouldn't need to change all call sites of `pktgen__push_default_iphdr`?

Reply:
Yes, I'll change that :+1:. Needing to adjust for every new unit-test that gets added during this PR's lifetime has started to get silly ... :)